llama_index/schema.py

```diff
@@ -21,20 +25,57 @@ def _validate_is_flat_dict(metadata_dict: dict) -> None:
             raise ValueError("Value must be one of (str, int, float, None)")
 
 
-@dataclass
-class BaseDocument(DataClassJsonMixin):
-    """Base document.
+class NodeRelationship(str, Enum):
+    """Node relationships used in `BaseNode` class.
 
-    Generic abstract interfaces that captures both index structs
-    as well as documents.
+    Attributes:
+        SOURCE: The node is the source document.
+        PREVIOUS: The node is the previous node in the document.
+        NEXT: The node is the next node in the document.
+        PARENT: The node is the parent node in the document.
+        CHILD: The node is a child node in the document.
 
     """
 
-    # TODO: consolidate fields from Document/IndexStruct into base class
-    text: Optional[str] = None
-    doc_id: Optional[str] = None
-    embedding: Optional[List[float]] = None
-    doc_hash: Optional[str] = None
+    SOURCE = auto()
+    PREVIOUS = auto()
+    NEXT = auto()
+    PARENT = auto()
+    CHILD = auto()
+
+
+class ObjectType(str, Enum):
+    DOCUMENT = auto()
+    TEXT = auto()
+    IMAGE = auto()
+    INDEX = auto()
+
+
+class RelatedNodeInfo(BaseModel):
+    node_id: str
+    metadata: Dict[str, Any] = Field(default_factory=dict)
+    hash: Optional[str] = None
+
+
+RELATED_NODE_TYPE = Union[RelatedNodeInfo, List[RelatedNodeInfo]]
+
+
+class BaseNode(BaseModel):
+    """Base node Object.
+
+    Generic abstract interface for retrievable nodes
+
+    """
+
+    _id: Optional[str] = Field(default=None, description="Unique ID of the node.")
+    embedding: Optional[List[float]] = Field(
+        default=None, description="Embedding of the node."
+    )
+    hash: Optional[str] = Field(default=None, description="Hash of the node content.")
+    weight: float = Field(
```

Comment:
I don't get what this does.

Reply:
Is the idea that the user want to surpress certain nodes?

Reply:
Yea it would basically weight the score of the retrieved nodes. Been a pretty common ask to weight some pieces of information more than others